File: VB/TempCollection.Module/Item.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo
Imports System.ComponentModel

Namespace TempCollection.Module
	<DefaultClassOptions, DefaultProperty("Id")> _
	Public Class Item
		Inherits BaseObject
		Private _id As String
		Private _quantity As Integer
		Private _Order As Order
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property Id() As String
			Get
				Return _id
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Id", _id, value)
			End Set
		End Property
		Public Property Quantity() As Integer
			Get
				Return _quantity
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Quantity", _quantity, value)
			End Set
		End Property
		<Association("RealAssociation")> _
		Public Property Order() As Order
			Get
				Return _Order
			End Get
			Set(ByVal value As Order)
				SetPropertyValue("Order", _Order, value)
			End Set
		End Property
	End Class
End Namespace

File: CS/TempCollection.Module/Item.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;
using System.ComponentModel;

namespace TempCollection.Module {
    [DefaultClassOptions]
    [DefaultProperty("Id")]
    public class Item : BaseObject {
        private string _id;
        private int _quantity;
        private Order _Order;
        public Item(Session session)
            : base(session) {
        }
        public string Id {
            get { return _id; }
            set {
                SetPropertyValue("Id", ref _id, value);
            }
        }
        public int Quantity {
            get { return _quantity; }
            set {
                SetPropertyValue("Quantity", ref _quantity, value);
            }
        }
        [Association("RealAssociation")]
        public Order Order {
            get { return _Order; }
            set { SetPropertyValue("Order", ref _Order, value); }
        }
    }
}